client/layout/guided-tours/wait.js

```diff
@@ -0,0 +1,14 @@
+const wait = ( condition, consequence, delay = 0 ) => {
+	if ( condition() ) {
+		consequence();
+		return;
+	}
+
+	window.setTimeout( wait.bind( null,
+		condition,
+		consequence,
+		delay ? delay * 2 : 1
+	), delay );
+};
+
+export default wait;
```

Comment:
I put this little guy inside the Guided Tours directory as a quick fix, but where do you guys think it really belongs?